ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -263,7 +271,7 @@ private void removeFiles(String location, ValidWriteIdList writeIdList, Compacti
     if (filesToDelete.size() < 1) {
       LOG.warn("Hmm, nothing to delete in the cleaner for directory " + location +
           ", that hardly seems right.");
-      return;
+      return false;
```

Comment:
I don't think this check helps. The dynamic partitioning case does a table level scan no? It might just find a single aborted dir in some of the old partitions that it can clean up, it does not mean, that it did not skip streaming deltas with aborted + commited txns.

Reply:
fixed by recording aborted writeIds

Reply:
@pvargacl , could you please take a 2nd look. Thank you!